src/neg_tests/lin_tests_common_dsl.ml

```diff
@@ -0,0 +1,86 @@
+open Lin_api
+
+(** ********************************************************************** *)
+(**                       Tests of a simple reference                      *)
+(** ********************************************************************** *)
+
+module Sut_int =
+  struct
+    let init () = ref 0
+    let get r = !r
+    let set r i = r:=i
+    let add r i = let old = !r in r:= i + old (* buggy: not atomic *)
+    let incr r = add r 1                      (* buggy: not atomic *)
+    let decr r = add r (-1)                   (* buggy: not atomic *)
+end
+
+module Sut_int64 =
+  struct
+    let init () = ref Int64.zero
+    let get r = !r
+    let set r i = r:=i
+    let add r i = let old = !r in r:= Int64.add i old (* buggy: not atomic *)
+    let incr r = add r Int64.one                      (* buggy: not atomic *)
+    let decr r = add r Int64.minus_one                (* buggy: not atomic *)
+end
+
+module Ref_int_spec : Lin_api.ApiSpec = struct
+  type t = int ref
+  let init () = Sut_int.init ()
+  let cleanup _ = ()
+  let api =
+    [ val_ "Sut_int.get"  Sut_int.get  (t @-> returning int);
+      val_ "Sut_int.set"  Sut_int.set  (t @-> int @-> returning unit);
+      val_ "Sut_int.add"  Sut_int.add  (t @-> int @-> returning unit);
+      val_ "Sut_int.incr" Sut_int.incr (t @-> returning unit);
+      val_ "Sut_int.decr" Sut_int.decr (t @-> returning unit);
+    ]
+  end
+
+module Ref_int64_spec : Lin_api.ApiSpec = struct
+  type t = int64 ref
+  let init () = Sut_int64.init ()
+  let cleanup _ = ()
+  let api =
+    [ val_ "Sut_int64.get"  Sut_int64.get  (t @-> returning int64);
+      val_ "Sut_int64.set"  Sut_int64.set  (t @-> int64 @-> returning unit);
+      val_ "Sut_int64.add"  Sut_int64.add  (t @-> int64 @-> returning unit);
+      val_ "Sut_int64.incr" Sut_int64.incr (t @-> returning unit);
+      val_ "Sut_int64.decr" Sut_int64.decr (t @-> returning unit);
+    ]
+  end
+
+module RT_int = Lin_api.Make(Ref_int_spec)
+module RT_int64 = Lin_api.Make(Ref_int64_spec)
+
+(** ********************************************************************** *)
+(**                  Tests of the buggy concurrent list CList              *)
+(** ********************************************************************** *)
+
+let int,int64 = nat_small,nat64_small
```

Comment:
One could move this line up to be in scope for all uses of `int` and `int64` in the file.
Alternatively, the definitions can be pushed into the individual `Spec` modules, so that it is clear to the reader that `int`/`int64` is overridden... :thinking:

Reply:
I've put the definitions in the respective `Spec` module. Closer to the reason is clearer I think.